DG-170 Notebook defect fixes

cdausmus opened the pull request:
- Fixed issue where validators couldn't edit transcriptions before they were validated. A button was available on the old notebook task history that was taken away in the new UI. This button has been returned.
 - Added a link to edit user for Site Admins
 - Updated filter and status labels to make it more clear what users are viewing.
 - Added a breakdown of users score into individual transcribed and validated scores.

cdausmus committed:
DG-170 Notebook fixes

 - Fixed issue where validators could no longer edit transcriptions
 - Added link to edit user for Site Admins
 - Updated status and filter labels to try and be less confusing

cdausmus committed:
Merge branch 'develop' into hotfix/6.2.1-notebook-fixes

cdausmus committed:
DG-170 Added score breakdown to score card

sbearcsiro commented on grails-app/controllers/au/org/ala/volunteer/UserController.groovy:
```diff
@@ -359,12 +361,21 @@ class UserController {
 
+        // Find out if the user is a validator for any of the projects/institutions of the tasks in the list
+        taskList.viewList.each { Map row ->
+            def taskProject = Project.get(row.projectId as long)
```
I believe `taskService.getNotebookTaskList` is only used in this controller action so these bits that hit the db should probably be moved inside the service method.

sbearcsiro commented on grails-app/taglib/au/org/ala/volunteer/VolunteerTagLib.groovy:
```diff
@@ -1350,4 +1350,13 @@ function notify() {
+
+    def externalLinkIcon = { attrs, body ->
+        out << "<a href=\"${attrs.href}\" target=\"_blank\" rel=\"noopener noreferrer\" title=\"${attrs.title ?: 'External link'}\">"
+        out << "<svg width=\"28\" height=\"28\" viewBox=\"0 0 28 28\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\">"
```
Could the SVG be put into an asset file?

sbearcsiro reviewed (commented):
Some nits picked

cdausmus committed:
DG-170 Fixes after review

 - Moved SVG to asset file
 - Moved check for project/institution validator to service

cdausmus committed:
DG-170 Code cleanup
